crates/bevy_reflect/src/tuple.rs

```diff
@@ -588,17 +588,17 @@ macro_rules! impl_reflect_tuple {
 
         impl<$($name: FromReflect),*> FromReflect for ($($name,)*)
         {
-            fn from_reflect(reflect: &dyn Reflect) -> Option<Self> {
+            fn from_reflect(reflect: &dyn Reflect) -> Result<Self, FromReflectError> {
                 if let ReflectRef::Tuple(_ref_tuple) = reflect.reflect_ref() {
-                    Some(
+                    Ok(
                         (
                             $(
-                                <$name as FromReflect>::from_reflect(_ref_tuple.field($index)?)?,
+                                <$name as FromReflect>::from_reflect(_ref_tuple.field($index).ok_or(FromReflectError{ from_type_name: reflect.type_name(), to_type: ReflectType::Tuple })?)?,
```

Comment:
Nit: Could we also split this up?
```suggestion
                                <$name as FromReflect>::from_reflect(
                                    _ref_tuple.field($index)
                                    .ok_or(FromReflectError {
                                        from_type_name: reflect.type_name(),
                                        to_type: ReflectType::Tuple
                                    })?
                                )?,
```